src/java/main/org/apache/zookeeper/server/quorum/Learner.java

```diff
@@ -230,15 +232,15 @@ protected void sockConnect(Socket sock, InetSocketAddress addr, int timeout)
     /**
      * Establish a connection with the Leader found by findLeader. Retries
      * until either initLimit time has elapsed or 5 tries have happened. 
-     * @param addr - the address of the Leader to connect to.
+     * @param leader - the QuorumServer elected leader
      * @throws IOException - if the socket connection fails on the 5th attempt
      * @throws ConnectException
      * @throws InterruptedException
      */
-    protected void connectToLeader(InetSocketAddress addr) 
-    throws IOException, ConnectException, InterruptedException {
-        sock = new Socket();        
-        sock.setSoTimeout(self.tickTime * self.initLimit);
+    protected void connectToLeader(QuorumServer leader)
+    throws IOException, InterruptedException, X509Exception {
+        QuorumX509Util quorumX509Util = new QuorumX509Util();
```

Comment:
We now create QuorumX509Util whenever connectToLeader is called, even if ssl is not enabled. Could this be improved so the SSL related code path is only involved when ssl is enabled?